File: selfstudy/js/students_sortasc.js
/*
task 19:
Students - SortAsc
You need to sort the values in ascending order for student test scores
*/

function sort_up_students(scores) {
  return [...scores].sort((a, b) => a - b);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: sort_up_students([10, 20, 30]) },
  { input: [50, 60, 70], expected: sort_up_students([50, 60, 70]) },
  { input: [1, 2, 3], expected: sort_up_students([1, 2, 3]) },
  { input: [100, 200], expected: sort_up_students([100, 200]) },
  { input: [], expected: sort_up_students([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sort_up_students', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sort_up_students(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sort_up_students(t.input) === t.expected || JSON.stringify(sort_up_students(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works

File: selfstudy/golang/students_sortasc.go
/*
task 219:
Students - SortAsc
You need to sort the values in ascending order for student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func Sort_upStudents(scores []int) []int {
	res := make([]int, len(scores))
	copy(res, scores)
	for i := 0; i < len(res); i++ {
		for j := i+1; j < len(res); j++ {
			if res[i] > res[j] { res[i], res[j] = res[j], res[i] }
		}
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Sort_upStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Sort_upStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Sort_upStudents([]int{1, 2, 3})},
		{[]int{100, 200}, Sort_upStudents([]int{100, 200})},
		{[]int{}, Sort_upStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := Sort_upStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
